### PR TITLE
fix: IWSLT2017

### PR DESCRIPTION
**Pull request by mrshu**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

* Add the IWSLT2017 BitextMining dataset, which contains 24 language pairs

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good. Not much to add here. Will you add points

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    "nl-en": ["nld-Latn", "eng-Latn"],
+    "nl-it": ["nld-Latn", "ita-Latn"],
+    "nl-ro": ["nld-Latn", "ron-Latn"],
+    "ro-en": ["ron-Latn", "eng-Latn"],
+    "ro-it": ["ron-Latn", "ita-Latn"],
+    "ro-nl": ["ron-Latn", "nld-Latn"],
+    "zh-en": ["cmn-Hans", "eng-Latn"],
+}
+
+_SPLITS = ["validation"]
```
Comment (KranthiGV) [inline mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:36]: Why are we not using `test` split?

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                name=f"iwslt2017-{lang}",
+                **self.metadata_dict["dataset"],
+            )
+
+        self.dataset_transform()
+        self.data_loaded = True
+
+    def dataset_transform(self):
+        def create_columns(row, lang):
+            l1, l2 = lang.split("-")
```
Comment (KranthiGV) [inline mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:106]: The data seems to be in JSON format `{ "ar": "قبل عدة سنوات، هنا في تيد، قدّم بيتر سكيلمان منافسة تصميم تسمى منافسة حلوى المارش مالو.", "en": "Several years ago here at TED, Peter Skillman introduced a design challenge called the marshmallow challenge." }` 

Why are we splitting on `-`?

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py (hunk 1 above)
Comment (mrshu) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:36]: @KranthiGV it was too big for direct use (8.5k samples). Happy to switch to the `test` split if that ends up being a better option.

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py (hunk 2 above)
Comment (mrshu) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:106]: @KranthiGV the `lang` here comes from the `_LANGUAGES` dictionary and is in the format `"zh-en"`, hence why the split on `-`.

File: docs/mmteb/points/727.jsonl [hunk 3]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "mrshu", "New dataset": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/727.jsonl:1]: 2 points pr. dataset and a 4 points bonus for adding a new language for bitext. I don't believe you do that so I think 2 points it is.

Comment (mrshu) [reply docs/mmteb/points/727.jsonl:1]: Understood, thanks!

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    booktitle = "Proceedings of the 14th International Conference on Spoken Language Translation",
+    month = dec # " 14-15",
+    year = "2017",
+    address = "Tokyo, Japan",
+    publisher = "International Workshop on Spoken Language Translation",
+    url = "https://aclanthology.org/2017.iwslt-1.1",
+    pages = "2--14",
+}
+""",
+        n_samples={"validation": 888},
```
[the remark this answers is not in the document]
Comment (mrshu) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:84]: Ah, I see. Yes, it's samples overall. Let me fix that.

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py (hunk 2 above)
Comment (KranthiGV) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:106]: Thanks! Missed it

**Review by KranthiGV — APPROVED** [review]
LGTM

File: mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py (hunk 1 above)
Comment (KranthiGV) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:36]: I think you can downsample like other datasets using `stratified_subsampling`

Comment (KranthiGV) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:36]: Example: `InternationalCitizenshipQuestionsLegalBenchClassification` class

Comment (mrshu) [reply mteb/tasks/BitextMining/multilingual/IWSLT2017BitextMinig.py:36]: Thanks @KranthiGV. That might be a good option for classification tasks (such as for instance `InternationalCitizenshipQuestionsLegalBenchClassification`) but the `stratified_subsampling` method needs a `label` parameters, and I am really not sure what `label` would we use in the context of a `BitextMining` task: https://github.com/embeddings-benchmark/mteb/blob/06a222b4aebbf43512658df4bcdd74d92a04a9fb/mteb/abstasks/AbsTask.py#L54

**Comment by mrshu** [comment]
Thanks again for all the help, this should be ready to be merged now.

**Pull request merged**